packages/headless/src/index.ts

```diff
@@ -223,11 +223,6 @@ export default async (opts: HeadlessOptions) => {
       pnpmVersion: opts.currentEngine.pnpmVersion,
     } as LockfileToDepGraphOptions
   )
-  if (filteredLockfile.packages) {
-    for (const skippedDepPath of Array.from(skipped)) {
-      delete filteredLockfile.packages[skippedDepPath]
```

Comment:
The performance regression was most likely caused by this line, since a `delete` statement in **V8** causes the object to no longer use fast properties, as it starts being treated similarly to a `Map` of `string`s, unless the object is considered to be a prototype object.

---

**See:** <https://github.com/sindresorhus/to-fast-properties/blob/master/index.js>

Reply:
I assumed that as well but I measured the time it took and it was nothing. The real reason was that the lockfile at `node_modules/.pnpm/lock.yaml` did not contain the skipped packages. pnpm compares that lockfile to the public one, which is at `pnpm-lock.yaml` and when they match, installation completes almost instantly. They did not match after the change done in #3032.